collector/container/konflux-slim.Dockerfile

```diff
@@ -1,55 +1,46 @@
 ARG BUILD_DIR=/build
 ARG CMAKE_BUILD_DIR=${BUILD_DIR}/cmake-build
 
+# Builder
 # TODO(ROX-20312): we can't pin image tag or digest because currently there's no mechanism to auto-update that.
-# TODO(ROX-20651): Use RHEL/ubi base image when entitlement is solved.
-# RPMs requiring entitlement: bpftool, cmake-3.18.2-9.el8, elfutils-libelf-devel, tbb-devel, c-ares-devel, jq-devel
-# FROM registry.access.redhat.com/ubi8/ubi:latest AS builder
-FROM quay.io/centos/centos:stream8 AS builder
+# TODO(ROX-20651): use content sets instead of subscription manager for access to RHEL RPMs once available.
+FROM registry.access.redhat.com/ubi8/ubi:latest AS ubi-normal
+FROM registry.access.redhat.com/ubi8/ubi:latest AS rpm-implanter-builder
 
-COPY . .
+COPY --from=ubi-normal / /mnt
+COPY ./.konflux /tmp/.konflux
 
 # TODO(ROX-20234): use hermetic builds when installing/updating RPMs becomes hermetic.
-RUN dnf upgrade -y --nobest \
-    && dnf -y install --nobest \
-        autoconf \
-        automake \
-        binutils-devel \
-        bison \
-        ca-certificates \
-        clang-15.0.7 \
-        cmake \
-        cracklib-dicts \
-        diffutils \
-        elfutils-libelf-devel \
-        file \
-        flex \
-        gcc \
-        gcc-c++ \
-        gdb \
-        gettext \
-        git \
-        glibc-devel \
-        libasan \
-        libubsan \
-        libcap-ng-devel \
-        libcurl-devel \
-        libtool \
-        libuuid-devel \
+RUN /tmp/.konflux/scripts/subscription-manager-bro.sh register /mnt && \
+    dnf -y --installroot=/mnt upgrade --nobest && \
+    dnf -y --installroot=/mnt install --nobest \
         make \
-        openssh-server \
-        openssl-devel \
-        patchutils \
-        passwd \
-        pkgconfig \
-        rsync \
-        tar \
-        unzip \
-        valgrind \
         wget \
-        which \
+        unzip \
```

Comment:
Updated the dependency list to https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/rhacs-1.0-rhel-8/distgit/containers/rhacs-collector-slim/Dockerfile.in?ref_type=heads. Previously it was the list of upstream dependencies because of the CentOS base image.